source/textUtils.py

```diff
@@ -12,15 +12,102 @@
 
 
-class WideStringOffsetConverter:
+class OffsetConverter:
+	decoded: str
+	
+	def __init__(self, text: str):
+		if not isinstance(text, str):
+			raise TypeError("Value must be of type str")
+		self.decoded: str = text
+
+	def __repr__(self):
+		return "{}({})".format(self.__class__.__name__, repr(self.decoded))
+
+	@property
+	def encodedStringLength(self) -> int:
+		"""Returns the length of the string in itssubclass-specific encoded representation."""
+		raise NotImplementedError
+
+	@property
+	def strLength(self) -> int:
+		"""Returns the length of the string in its pythonic string representation."""
+		return len(self.decoded)
+
+	@abstractmethod
+	def strToEncodedOffsets(
+			self,
+			strStart: int,
+			strEnd: int | None = None,
+			raiseOnError: bool = False,
+	) -> int | Tuple[int, int]:
+		"""
+		This method takes two offsets from the str representation
+		of the string the object is initialized with, and converts them to subclass-specific encoded string offsets.
+		@param strStart: The start offset in the str representation of the string.
+		@param strEnd: The end offset in the str representation of the string.
+			This offset is exclusive.
+		@param raiseOnError: Raises an IndexError when one of the given offsets
+			exceeds L{strLength} or is lower than zero.
+			If C{False}, the out of range offset will be bounded to the range of the string.
+		@raise ValueError: if strEnd < strStart
+		"""
+		if strEnd is not None and strEnd < strStart:
+			raise ValueError(
+				"strEnd=%d must be greater than or equal to strStart=%d"
+				% (strEnd, strStart)
+			)
+		if strStart < 0 or strStart > self.strLength:
+			if raiseOnError:
+				raise IndexError("str start index out of range")
+			strStart = max(0, min(strStart, self.strLength))
+		if strEnd is not None and (strEnd < 0 or strEnd > self.strLength):
+			if raiseOnError:
+				raise IndexError("str end index out of range")
+			strEnd = max(0, min(strEnd, self.strLength))
+
+	@abstractmethod
+	def encodedToStrOffsets(
+			self,
+			encodedStart: int,
+			encodedEnd: int | None = None,
+			raiseOnError: bool = False,
+	) -> int | Tuple[int, int]:
+		r"""
+		This method takes two offsets from subclass-specific encoded string representation
+		of the string the object is initialized with, and converts them to str offsets.
+		@param encodedStart: The start offset in the wide character representation of the string.
+		@param encodedEnd: The end offset in the wide character representation of the string.
+			This offset is exclusive.
+		@param raiseOnError: Raises an IndexError when one of the given offsets
+			exceeds L{encodedStringLength} or is lower than zero.
+			If C{False}, the out of range offset will be bounded to the range of the string.
+		@raise ValueError: if wideStringEnd < wideStringStart
+		"""
+		if encodedEnd is not None and encodedEnd < encodedStart:
+			raise ValueError(
+				"encodedEnd=%d must be greater than or equal to encodedStart=%d"

```

Comment:
Same as above W.R.t. usage of f-strings.

Reply:
when refactoring legacy code, please modernise it. Otherwise it will never get fixed.

Reply:
Done.